libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -612,13 +612,38 @@ bool DeclarationRegistrationHelper::visitNode(ASTNode& _node)
 
 	if (auto* declaration = dynamic_cast<Declaration*>(&_node))
 		registerDeclaration(*declaration);
+
+	// Setting the canonicalname before ScopeOpener, so that m_currentScope is not same as the
+	// current _node
+	if (auto* annotation = dynamic_cast<TypeDeclarationAnnotation*>(&_node.annotation()))
+	{
+		string canonicalName;
+
+		if (auto decl = dynamic_cast<Declaration const*>(&_node))
```

Comment:
Can this really be false? If yes, I think we should do something, it would mean that the name of a type declaration does not contain its own name, but the name of a super scope...